proto/catalog.proto

```diff
@@ -67,10 +75,10 @@ message Sink {
   repeated int32 distribution_key = 8;
   // pk_indices of the corresponding materialize operator's output.
   repeated int32 stream_key = 9;
-  bool append_only = 10;
-  uint32 owner = 11;
-  map<string, string> properties = 12;
-  string definition = 13;
+  uint32 owner = 10;
+  map<string, string> properties = 11;
+  string definition = 12;
+  SinkType sink_type = 13;
```

Comment:
I think we need not change too many fields here and the field number does not need to be compacted and ordered in fact....